epochStart/metachain/systemSCs.go

```diff
@@ -343,6 +354,48 @@ func (s *systemSCProcessor) unStakeOneNode(blsKey []byte, epoch uint32) error {
 	return nil
 }
 
+func (s *systemSCProcessor) updateDelegationContracts(mapOwnerKeys map[string][][]byte) error {
+	sortedDelegationsSCs := make([]string, 0, len(mapOwnerKeys))
+	for address := range mapOwnerKeys {
+		if s.shardCoordinator.ComputeId([]byte(address)) != core.MetachainShardId {
+			continue
+		}
+		sortedDelegationsSCs = append(sortedDelegationsSCs, address)
+	}
+
+	sort.Slice(sortedDelegationsSCs, func(i, j int) bool {
+		return sortedDelegationsSCs[i] < sortedDelegationsSCs[j]
+	})
+
+	for _, address := range sortedDelegationsSCs {
```

Comment:
I think we should have a limit or a sort of limit because multiple calls in a for loop at the end of epoch might result in a metachain stuck situation when it does not have enough time to iterate all delegation SCs

Reply:
I think there should be no big deal here and we have enough processing time.

Reply:
Initial tests when we had similar process proved otherwise

Reply:
this is somewhat hard to reach. to have 1000 calls here. you need 1000 delegation contracts where you have unDelegate in the same epoch for more values.